packages/generator/src/action-function-import/request-builder-arguments.ts

```diff
@@ -9,13 +10,32 @@ export function getRequestBuilderArgumentsBase(
   actionFunctionImport: VdmFunctionImport | VdmActionImport,
   service: VdmServiceMetadata
 ): string[] {
+  const transformer = getTransformer(actionFunctionImport, service);
   return [
     `'${service.servicePath}'`,
     `'${actionFunctionImport.originalName}'`,
-    `(data) => ${responseTransformerFunctionName(
-      actionFunctionImport.returnType,
-      service.oDataVersion
-    )}(data, ${actionFunctionImport.returnType.builderFunction})`,
+    transformer,
     'params'
   ];
 }
+
+function getTransformer(
+  actionFunctionImport: VdmFunctionImport | VdmActionImport,
+  service: VdmServiceMetadata
+): string {
+  if (
+    actionFunctionImport.returnType.returnTypeCategory ===
+    VdmReturnTypeCategory.ENTITY_NOT_DESERIALIZABLE
+  ) {
+    return `(data) => throwErrorWhenReturnTypeIsUnionType(data, '${actionFunctionImport.originalName}')`;
```

Comment:
Minor: Maybe it would be a good idea to not go through the core to call `throwErrorWhenReturnTypeIsUnionType` but just throw an error here directly. I don't have any good arguments for this in mind, though.